internal/testing/runchecks.sh

```diff
@@ -26,10 +26,41 @@ if [[ $# -gt 0 ]]; then
   exit 64
 fi
 
-result=0
+# The following logic lets us skip the (lengthy) installation process and tests
+# in some cases where the PR carries trivial changes that don't affect the code
+# (such as documentation-only).
+if [[ -z "$TRAVIS_BRANCH" || -z "$TRAVIS_PULL_REQUEST_SHA" ]]; then
+  echo "TRAVIS_BRANCH and TRAVIS_PULL_REQUEST_SHA environment variables must be set." 1>&2
+  exit 1
+fi
+
+tmpfile=$(mktemp)
+function cleanup() {
+  rm -rf "$tmpfile"
+}
+trap cleanup EXIT
+
+mergebase="$(git merge-base -- "$TRAVIS_BRANCH" "$TRAVIS_PULL_REQUEST_SHA")"
+git diff --name-only "$mergebase" "$TRAVIS_PULL_REQUEST_SHA" -- > $tmpfile
```

Comment:
You can do without the temporary file by assigning to another variable:

```suggestion
has_code_changes=0
git diff --name-only "$mergebase" "$TRAVIS_PULL_REQUEST_SHA" -- | grep -v ^internal/website > /dev/null || has_code_changes=1
```

Reply:
Keeping it more explicit with a temp file, as discussed.
If you prefer to factor the pattern out into a variable, I can do that. Or we can do it in a followup PR that may actually want to add patterns, where it will be easier to see what the commonailty is.